File: internal/content/testdata/bench/java/minimum-pages-allocation.java
class Solution {
    public long min_max_pages(long[] pages, long num_students) {
        if (num_students > pages.length) return -1;
        long lo = 0, hi = 0;
        for (long p : pages) {
            if (p > lo) lo = p;
            hi += p;
        }
        long result = hi;
        while (lo <= hi) {
            long mid = (lo + hi) / 2;
            if (canAllocate(pages, num_students, mid)) {
                result = mid;
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }
        return result;
    }

    private boolean canAllocate(long[] pages, long numStudents, long maxPages) {
        long students = 1, current = 0;
        for (long p : pages) {
            if (p > maxPages) return false;
            if (current + p > maxPages) {
                students++;
                current = p;
            } else {
                current += p;
            }
        }
        return students <= numStudents;
    }
}

File: internal/content/testdata/bench/typescript/minimum-pages-allocation.ts
function min_max_pages(pages: number[], num_students: number): number {
    if (num_students > pages.length) return -1;

    function can_allocate(max_pages: number): boolean {
        let students = 1;
        let current = 0;
        for (const p of pages) {
            if (p > max_pages) return false;
            if (current + p > max_pages) {
                students++;
                current = p;
            } else {
                current += p;
            }
        }
        return students <= num_students;
    }

    let lo = Math.max(...pages);
    let hi = pages.reduce((a, b) => a + b, 0);
    let result = hi;
    while (lo <= hi) {
        const mid = Math.floor((lo + hi) / 2);
        if (can_allocate(mid)) {
            result = mid;
            hi = mid - 1;
        } else {
            lo = mid + 1;
        }
    }
    return result;
}
